ls1intum/Artemis
# Feature/import text exercises

### Checklist

### Motivation and Context
This PR is part of a bigger feature to allow different Exercises to be imported. In this PR the focus is to allow for text exercises to be imported. This works for import Text Exercises: 
- Course to Course
- Course to Exam
- Exam to Course
- Exam to Exam

### Description
Client:
- [x] Import for course text exercises
- [x] Import for exam text exercises
- [x] Create import text exercises modal
- [x] Added "Import text exercise" button to exercise group component  
- [x] Added "Import text exercise" button to course-management component  
- [x] Fix Button placement for the exercise-group component
- [x] Added jest client tests

Server:
- [x] Setup TextExerciseImportService
- [x] Add search queries for text exercises for admin and instructors based on their courses
- [x] Add search queries for text exercises for instructors based on their courses
- [x] Add additional authorization check to GET programming exercise route. Only instructors and admins should be allowed to access an exam programming exercise

### Steps for Testing
Test the import feature for all import directions mentioned above.

To test the import feature: 
1. Log in to Artemis
2. Navigate to Course Administration
3. Click on import Text Exercise
4. Select the Exercise you want to import
5. Change a few attributes of the Exercise and press import. 
6. All changes should be saved in a new exercise, including the example submission and Assessments. 
7. For good measure, delete the original exercise and make sure everything still works for the imported Exercise. 

### Screenshots
- **Text Exercise Component: Add Import Button**
![image](https://user-images.githubusercontent.com/15110960/85055425-d1704880-b19d-11ea-93c4-defec1cc7f14.png)

- **Import Text Exercise Modal** 
![image](https://user-images.githubusercontent.com/15110960/85224793-07dfda80-b3cd-11ea-94d2-eb40b3ac0083.png)

- **Text Exercise Update Component: Adapt for importing exercises**
![image](https://user-images.githubusercontent.com/15110960/85055607-1ac09800-b19e-11ea-845f-fac294a6faa0.png)
![image](https://user-images.githubusercontent.com/15110960/85055637-24e29680-b19e-11ea-8cce-318fbd385686.png)

- **Import Text Exercise Button for Exercise Group**
![image](https://user-images.githubusercontent.com/15110960/85224783-f565a100-b3cc-11ea-8a31-cfd55bfe1c0f.png)
